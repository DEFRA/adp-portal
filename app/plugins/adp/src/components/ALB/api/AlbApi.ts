import {
  ArmsLengthBody,
  CreateArmsLengthBodyRequest,
  UpdateArmsLengthBodyRequest,
} from '@internal/plugin-adp-common';
import { createApiRef } from '@backstage/core-plugin-api';

export const armsLengthBodyApiRef = createApiRef<ArmsLengthBodyApi>({
  id: 'plugin.adp.armslengthbodyapi',
});

export interface ArmsLengthBodyApi {
  getArmsLengthBodies(): Promise<ArmsLengthBody[]>;
  updateArmsLengthBody(
    data: UpdateArmsLengthBodyRequest,
  ): Promise<ArmsLengthBody[]>;
  createArmsLengthBody(
    data: CreateArmsLengthBodyRequest,
  ): Promise<ArmsLengthBody[]>;
  getArmsLengthBodyNames(): Promise<Record<string, string>>;
}
