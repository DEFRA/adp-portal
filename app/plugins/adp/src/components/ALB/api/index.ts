export { ArmsLengthBodyClient } from './AlbClient'
export type {ArmsLengthBodyApi} from './AlbApi'
