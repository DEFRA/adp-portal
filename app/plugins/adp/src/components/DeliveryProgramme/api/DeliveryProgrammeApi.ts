import {
  DeliveryProgramme,
  DeliveryProgrammeAdmin,
} from '@internal/plugin-adp-common';
import { createApiRef } from '@backstage/core-plugin-api';

export const DeliveryProgrammeApiRef = createApiRef<DeliveryProgramme>({
  id: 'plugin.adp.deliveryprogrammeapi',
});

export interface DeliveryProgrammeApi {
  getDeliveryProgrammes(): Promise<DeliveryProgramme[]>;
  getDeliveryProgrammeById(id: string): Promise<DeliveryProgramme>;
  updateDeliveryProgramme(data: any): Promise<DeliveryProgramme[]>;
  createDeliveryProgramme(data: any): Promise<DeliveryProgramme[]>;
  getDeliveryProgrammeAdmins(): Promise<DeliveryProgrammeAdmin[]>;
}
