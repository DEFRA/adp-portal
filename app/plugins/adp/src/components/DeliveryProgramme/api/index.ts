export { DeliveryProgrammeClient } from './DeliveryProgrammeClient';
export type { DeliveryProgrammeApi } from './DeliveryProgrammeApi';
