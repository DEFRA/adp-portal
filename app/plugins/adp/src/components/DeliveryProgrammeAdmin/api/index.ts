export { deliveryProgrammeAdminApiRef } from './DeliveryProgrammeAdminApi';
export { DeliveryProgrammeAdminApiClient } from './DeliveryProgrammeAdminClient';
