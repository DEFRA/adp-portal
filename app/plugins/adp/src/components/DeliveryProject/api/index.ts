export * from './DeliveryProjectClient';
export * from './DeliveryProjectApi';
