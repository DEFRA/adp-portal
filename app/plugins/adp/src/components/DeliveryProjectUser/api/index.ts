export {
  deliveryProjectUserApiRef,
  type DeliveryProjectUserApi,
} from './DeliveryProjectUserApi';
export { DeliveryProjectUserClient } from './DeliveryProjectUserClient';
