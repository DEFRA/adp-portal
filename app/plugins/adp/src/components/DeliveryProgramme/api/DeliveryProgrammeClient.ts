import { DeliveryProgrammeApi } from './DeliveryProgrammeApi';
import {
  DeliveryProgramme,
  ProgrammeManager,
} from '@internal/plugin-adp-common';

import { DiscoveryApi, FetchApi } from '@backstage/core-plugin-api';
import { ResponseError } from '@backstage/errors';

export class DeliveryProgrammeClient implements DeliveryProgrammeApi {
  private discoveryApi: DiscoveryApi;
  private fetchApi: FetchApi;

  constructor(discoveryApi: DiscoveryApi, fetchApi: FetchApi) {
    this.discoveryApi = discoveryApi;
    this.fetchApi = fetchApi;
  }

  private async getApiUrl(): Promise<string> {
    return `${await this.discoveryApi.getBaseUrl('adp')}/deliveryProgramme`;
  }

  private async getManagerApiUrl(): Promise<string> {
    return `${await this.discoveryApi.getBaseUrl('adp')}/programmeManager`;
  }

  async getDeliveryProgrammes(): Promise<DeliveryProgramme[]> {
    try {
      const url = await this.getApiUrl();

      const albNamesUrl = `${await this.discoveryApi.getBaseUrl(
        'adp',
      )}/armslengthbodynames`;

      const [deliveryProgrammesResponse, albNamesResponse] = await Promise.all([
        this.fetchApi.fetch(url),
        this.fetchApi.fetch(albNamesUrl),
      ]);

      if (!deliveryProgrammesResponse.ok || !albNamesResponse.ok) {
        throw new Error('Failed to fetch data');
      }

      const [deliveryProgrammes, albNamesMapping] = await Promise.all([
        deliveryProgrammesResponse.json(),
        albNamesResponse.json(),
      ]);

      const deliveryProgrammesWithNames = deliveryProgrammes.map(
        (programme: { arms_length_body_id: string | number }) => ({
          ...programme,

          arms_length_body_id_name:
            albNamesMapping[programme.arms_length_body_id] ||
            'Unknown ALB Name',
        }),
      );

      return deliveryProgrammesWithNames;
    } catch (error) {
      throw new Error(`Failed to fetch Delivery Programmes`);
    }
  }

  async createDeliveryProgramme(data: any): Promise<DeliveryProgramme[]> {
    const url = await this.getApiUrl();
    const response = await this.fetchApi.fetch(url, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(data),
    });

    if (!response.ok) {
      throw await ResponseError.fromResponse(response);
    }

    return response.json();
  }

  async updateDeliveryProgramme(data: any): Promise<DeliveryProgramme[]> {
    const url = await this.getApiUrl();

    const response = await this.fetchApi.fetch(url, {
      method: 'PATCH',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(data),
    });

    if (!response.ok) {
      throw await ResponseError.fromResponse(response);
    }

    const updatedData: DeliveryProgramme[] = await response.json();
    return updatedData;
  }

  async getDeliveryProgrammeById(id: string): Promise<DeliveryProgramme> {
    try {
      const url = await this.getApiUrl();
      const response = await this.fetchApi.fetch(`${url}/${id}`);
      if (!response.ok) {
        throw await ResponseError.fromResponse(response);
      }
      return await response.json();
    } catch (error) {
      throw new Error(`Failed to fetch Delivery Programme by ID`);
    }
  }

  async getProgrammeManagers(): Promise<ProgrammeManager[]> {
    try {
      const url = await this.getManagerApiUrl();
      const response = await this.fetchApi.fetch(url);
      if (!response.ok) {
        throw await ResponseError.fromResponse(response);
      }
      return await response.json();
    } catch (error) {
      throw new Error(`Failed to fetch Delivery Programme Managers`);
    }
  }
}

export type { DeliveryProgrammeApi };
