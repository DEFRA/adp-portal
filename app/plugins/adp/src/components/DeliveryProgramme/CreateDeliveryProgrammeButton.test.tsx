import { AlertApi, alertApiRef } from '@backstage/core-plugin-api';
import React from 'react';
import { DeliveryProgrammeApi, deliveryProgrammeApiRef } from './api';
import { render as testRender, waitFor } from '@testing-library/react';
import { TestApiProvider } from '@backstage/test-utils';
import {
  CreateDeliveryProgrammeButton,
  CreateDeliveryProgrammeButtonProps,
} from './CreateDeliveryProgrammeButton';
import userEvent from '@testing-library/user-event';
import {
  DeliveryProgrammeFields,
  DeliveryProgrammeFormFields,
  emptyForm,
} from './DeliveryProgrammeFormFields';
import { act } from 'react-dom/test-utils';
import { ValidationError as IValidationError } from '@internal/plugin-adp-common';
import { ValidationError } from '../../utils';

const usePermission: jest.MockedFn<
  typeof import('@backstage/plugin-permission-react').usePermission
> = jest.fn();
const DialogForm: jest.MockedFn<
  typeof import('../../utils/DialogForm').DialogForm
> = jest.fn();

const fields: DeliveryProgrammeFields = {
  alias: 'abc',
  arms_length_body_id: 'def',
  delivery_programme_code: 'ghi',
  description: 'jkl',
  title: 'mno',
  url: 'pqr',
};

beforeEach(() => {
  jest.resetAllMocks();
});

describe('CreateDeliveryProgrammeButton', () => {
  it('Should not render when the user is not allowed to create delivery programmes', async () => {
    const { mockAlertApi, mockProgrammeApi, render } = setup();
    usePermission.mockReturnValue({ allowed: false, loading: false });

    const { result } = await render({ content: 'My button' });

    expect(result.baseElement).toMatchSnapshot();
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
    expect(mockProgrammeApi.createDeliveryProgramme).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammeAdmins).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammeById).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammes).not.toHaveBeenCalled();
    expect(mockProgrammeApi.updateDeliveryProgramme).not.toHaveBeenCalled();
  });
  it('Should only render a button initially', async () => {
    const { mockAlertApi, mockProgrammeApi, render } = setup();
    usePermission.mockReturnValue({ allowed: true, loading: false });

    const { result } = await render({ content: 'My button' });

    expect(result.baseElement).toMatchSnapshot();
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
    expect(mockProgrammeApi.createDeliveryProgramme).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammeAdmins).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammeById).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammes).not.toHaveBeenCalled();
    expect(mockProgrammeApi.updateDeliveryProgramme).not.toHaveBeenCalled();
  });
  it('Should render the dialog when the button is clicked', async () => {
    const { mockAlertApi, mockProgrammeApi, render } = setup();
    usePermission.mockReturnValue({ allowed: true, loading: false });
    DialogForm.mockReturnValue(<span>This is a dialog!</span>);

    const { result } = await render({ content: 'My button' });
    await userEvent.click(
      result.getByTestId('create-delivery-programme-button'),
    );

    expect(result.baseElement).toMatchSnapshot();
    expect(DialogForm.mock.calls).toHaveLength(1);
    const formProps = DialogForm.mock.calls[0][0];
    expect({
      open: formProps.open,
      renderFields: formProps.renderFields,
      confirm: formProps.confirm,
      cancel: formProps.cancel,
      defaultValues: formProps.defaultValues,
      disabled: formProps.disabled,
      validate: formProps.validate,
    }).toMatchObject({
      open: undefined,
      renderFields: DeliveryProgrammeFormFields,
      confirm: 'Create',
      cancel: undefined,
      defaultValues: emptyForm,
      disabled: undefined,
      validate: undefined,
    });
    expect(formProps.submit).toBeDefined();
    expect(formProps.completed).toBeDefined();
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
    expect(mockProgrammeApi.createDeliveryProgramme).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammeAdmins).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammeById).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammes).not.toHaveBeenCalled();
    expect(mockProgrammeApi.updateDeliveryProgramme).not.toHaveBeenCalled();
  });
  it('Should close the form when the form is cancelled.', async () => {
    const { mockAlertApi, mockProgrammeApi, render } = setup();
    usePermission.mockReturnValue({ allowed: true, loading: false });
    DialogForm.mockReturnValue(<span>This is a dialog!</span>);

    const { result } = await render({ content: 'My button' });
    await userEvent.click(
      result.getByTestId('create-delivery-programme-button'),
    );

    expect(result.baseElement).toMatchSnapshot('Before cancel');
    expect(DialogForm.mock.calls).toHaveLength(1);
    const formProps = DialogForm.mock.calls[0][0];
    act(() => formProps.completed(undefined));
    await waitFor(() =>
      expect(result.queryByText('This is a dialog!')).toBeNull(),
    );
    expect(result.baseElement).toMatchSnapshot('After cancel');
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
    expect(mockProgrammeApi.createDeliveryProgramme).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammeAdmins).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammeById).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammes).not.toHaveBeenCalled();
    expect(mockProgrammeApi.updateDeliveryProgramme).not.toHaveBeenCalled();
  });
  it('Should call onCreated when the form closes with a value.', async () => {
    const { mockAlertApi, mockProgrammeApi, render } = setup();
    usePermission.mockReturnValue({ allowed: true, loading: false });
    DialogForm.mockReturnValue(<span>This is a dialog!</span>);
    const onCreated: jest.MockedFn<
      Exclude<CreateDeliveryProgrammeButtonProps['onCreated'], undefined>
    > = jest.fn();

    const { result } = await render({ content: 'My button', onCreated });
    await userEvent.click(
      result.getByTestId('create-delivery-programme-button'),
    );

    expect(result.baseElement).toMatchSnapshot('Before complete');
    expect(DialogForm.mock.calls).toHaveLength(1);
    expect(onCreated).not.toHaveBeenCalled();
    const formProps = DialogForm.mock.calls[0][0];
    act(() => formProps.completed(fields));
    await waitFor(() =>
      expect(result.queryByText('This is a dialog!')).toBeNull(),
    );
    expect(result.baseElement).toMatchSnapshot('After complete');
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
    expect(onCreated).toHaveBeenCalledTimes(1);
    expect(mockProgrammeApi.createDeliveryProgramme).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammeAdmins).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammeById).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammes).not.toHaveBeenCalled();
    expect(mockProgrammeApi.updateDeliveryProgramme).not.toHaveBeenCalled();
  });
  it('Should call the api when submitting.', async () => {
    const { mockAlertApi, mockProgrammeApi, render } = setup();
    usePermission.mockReturnValue({ allowed: true, loading: false });
    DialogForm.mockReturnValue(<span>This is a dialog!</span>);

    const { result } = await render({ content: 'My button' });
    await userEvent.click(
      result.getByTestId('create-delivery-programme-button'),
    );

    expect(result.baseElement).toMatchSnapshot();
    expect(DialogForm.mock.calls).toHaveLength(1);
    const formProps = DialogForm.mock.calls[0][0];
    expect(mockProgrammeApi.createDeliveryProgramme).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammeAdmins).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammeById).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammes).not.toHaveBeenCalled();
    expect(mockProgrammeApi.updateDeliveryProgramme).not.toHaveBeenCalled();
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();

    const submitResult = await formProps.submit(fields);
    expect(submitResult).toMatchObject({ type: 'success' });
    expect(mockProgrammeApi.createDeliveryProgramme.mock.calls).toMatchObject([
      [fields],
    ]);
    expect(mockProgrammeApi.getDeliveryProgrammeAdmins).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammeById).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammes).not.toHaveBeenCalled();
    expect(mockProgrammeApi.updateDeliveryProgramme).not.toHaveBeenCalled();
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post.mock.calls).toMatchObject([
      [
        {
          message: 'Delivery Programme created successfully.',
          severity: 'success',
          display: 'transient',
        },
      ],
    ]);
  });
  it('Should catch ValidationErrors when submitting.', async () => {
    const { mockAlertApi, mockProgrammeApi, render } = setup();
    usePermission.mockReturnValue({ allowed: true, loading: false });
    DialogForm.mockReturnValue(<span>This is a dialog!</span>);
    const validationErrors: IValidationError[] = [
      {
        path: 'abc',
        error: {
          message: '123',
        },
      },
    ];
    mockProgrammeApi.createDeliveryProgramme.mockRejectedValueOnce(
      new ValidationError(validationErrors),
    );

    const { result } = await render({ content: 'My button' });
    await userEvent.click(
      result.getByTestId('create-delivery-programme-button'),
    );

    expect(result.baseElement).toMatchSnapshot();
    expect(DialogForm.mock.calls).toHaveLength(1);
    const formProps = DialogForm.mock.calls[0][0];
    expect(mockProgrammeApi.createDeliveryProgramme).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammeAdmins).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammeById).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammes).not.toHaveBeenCalled();
    expect(mockProgrammeApi.updateDeliveryProgramme).not.toHaveBeenCalled();

    const submitResult = await formProps.submit(fields);
    expect(submitResult).toMatchObject({
      type: 'validationError',
      errors: validationErrors,
    });
    expect(mockProgrammeApi.createDeliveryProgramme.mock.calls).toMatchObject([
      [fields],
    ]);
    expect(mockProgrammeApi.getDeliveryProgrammeAdmins).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammeById).not.toHaveBeenCalled();
    expect(mockProgrammeApi.getDeliveryProgrammes).not.toHaveBeenCalled();
    expect(mockProgrammeApi.updateDeliveryProgramme).not.toHaveBeenCalled();
    expect(mockAlertApi.alert$).not.toHaveBeenCalled();
    expect(mockAlertApi.post).not.toHaveBeenCalled();
  });
});

function setup() {
  const mockAlertApi: jest.Mocked<AlertApi> = {
    alert$: jest.fn(),
    post: jest.fn(),
  };
  const mockProgrammeApi: jest.Mocked<DeliveryProgrammeApi> = {
    createDeliveryProgramme: jest.fn(),
    getDeliveryProgrammeAdmins: jest.fn(),
    getDeliveryProgrammeById: jest.fn(),
    updateDeliveryProgramme: jest.fn(),
    getDeliveryProgrammes: jest.fn(),
  };

  return {
    mockAlertApi,
    mockProgrammeApi,
    async render(props: CreateDeliveryProgrammeButtonProps) {
      const result = testRender(
        <TestApiProvider
          apis={[
            [alertApiRef, mockAlertApi],
            [deliveryProgrammeApiRef, mockProgrammeApi],
          ]}
        >
          <CreateDeliveryProgrammeButton {...props} />
        </TestApiProvider>,
      );
      await waitFor(() => expect(result.baseElement).not.toBeEmptyDOMElement());
      return { result };
    },
  };
}

jest.mock(
  '@backstage/plugin-permission-react',
  () =>
    ({
      get usePermission() {
        return usePermission;
      },
      get IdentityPermissionApi(): never {
        throw new Error('Not mocked');
      },
      get PermissionedRoute(): never {
        throw new Error('Not mocked');
      },
      get RequirePermission(): never {
        throw new Error('Not mocked');
      },
      get permissionApiRef(): never {
        throw new Error('Not mocked');
      },
    } satisfies typeof import('@backstage/plugin-permission-react')),
);

jest.mock(
  '../../utils/DialogForm',
  () =>
    ({
      get DialogForm() {
        return DialogForm as typeof import('../../utils/DialogForm').DialogForm;
      },
    } satisfies typeof import('../../utils/DialogForm')),
);
