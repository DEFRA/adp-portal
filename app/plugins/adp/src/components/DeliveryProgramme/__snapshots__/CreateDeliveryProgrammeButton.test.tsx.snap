// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateDeliveryProgrammeButton Should call onCreated when the form closes with a value.: After complete 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="My button"
      data-testid="create-delivery-programme-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
  </div>
</body>
`;

exports[`CreateDeliveryProgrammeButton Should call onCreated when the form closes with a value.: Before complete 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="My button"
      data-testid="create-delivery-programme-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <span>
      This is a dialog!
    </span>
  </div>
</body>
`;

exports[`CreateDeliveryProgrammeButton Should call the api when submitting. 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="My button"
      data-testid="create-delivery-programme-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <span>
      This is a dialog!
    </span>
  </div>
</body>
`;

exports[`CreateDeliveryProgrammeButton Should call the api when submitting.: Before complete 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="My button"
      data-testid="create-delivery-programme-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`CreateDeliveryProgrammeButton Should catch ValidationErrors when submitting. 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="My button"
      data-testid="create-delivery-programme-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <span>
      This is a dialog!
    </span>
  </div>
</body>
`;

exports[`CreateDeliveryProgrammeButton Should close the form when the form is cancelled.: After cancel 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="My button"
      data-testid="create-delivery-programme-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
  </div>
</body>
`;

exports[`CreateDeliveryProgrammeButton Should close the form when the form is cancelled.: Before cancel 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="My button"
      data-testid="create-delivery-programme-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <span>
      This is a dialog!
    </span>
  </div>
</body>
`;

exports[`CreateDeliveryProgrammeButton Should not render when the user is not allowed to create delivery programmes 1`] = `
<body>
  <div />
</body>
`;

exports[`CreateDeliveryProgrammeButton Should only render a button initially 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="My button"
      data-testid="create-delivery-programme-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</body>
`;

exports[`CreateDeliveryProgrammeButton Should render the dialog when the button is clicked 1`] = `
<body>
  <div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      content="My button"
      data-testid="create-delivery-programme-button"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      />
      <span
        class="MuiTouchRipple-root"
      >
        <span
          class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
    <span>
      This is a dialog!
    </span>
  </div>
</body>
`;
